### PR TITLE
Add a doc comment for OS.errorCode

### PR DESCRIPTION
**Pull request by daviditen**
Add a doc comment for OS.errorCode derived from the comment for
SysBasic.syserr.

Signed-off-by: David Iten <daviditen@users.noreply.github.com>

### CODE REVIEW

File: modules/standard/OS.chpl [hunk 1]
```diff
@@ -956,6 +956,25 @@ module OS {
...
+   value of a :type:`CTypes.c_int` or another :type:`errorCode`. In addition,
+   :type:`errorCode` can be checked directly in an if statement like so:
+
+   .. code-block:: chapel
+
+     var err: errorCode;
+     if err then do writeln("err contains an error, ie err != ENOERR");
+     if !err then do writeln("err does not contain an error; err == ENOERR");
+
+   When an :type:`errorCode` formal has default intent, the actual is copied to
```
Comment (dlongnecke-cray) [inline modules/standard/OS.chpl:972]: Consider: "The default intent for a formal of type `errorCode` is `const in`? I think that communicates the same thing.

File: modules/standard/OS.chpl [hunk 2]
```diff
@@ -956,6 +956,25 @@ module OS {
...
+   :type:`CTypes.c_int` or to another :type:`errorCode`. An :type:`errorCode`
+   can be cast to or from a :type:`CTypes.c_int`. It can be assigned the
+   value of a :type:`CTypes.c_int` or another :type:`errorCode`. In addition,
+   :type:`errorCode` can be checked directly in an if statement like so:
+
+   .. code-block:: chapel
+
+     var err: errorCode;
+     if err then do writeln("err contains an error, ie err != ENOERR");
+     if !err then do writeln("err does not contain an error; err == ENOERR");
```
Comment (dlongnecke-cray) [inline modules/standard/OS.chpl:970]: Maybe just make this into an `if/else`?

File: modules/standard/OS.chpl [hunk 3]
```diff
@@ -956,6 +956,25 @@ module OS {
...
+
+   When an :type:`errorCode` formal has default intent, the actual is copied to
+   the formal upon a function call and the formal cannot be assigned within
+   the function.
+
+   The default value of the :type:`errorCode` type is undefined.
```
Comment (dlongnecke-cray) [inline modules/standard/OS.chpl:976]: I don't understand why we wouldn't just have the default value be `ENOERR`? Can we not define a default constructor or something? (Is that not in the language yet?)

Comment (daviditen) [reply modules/standard/OS.chpl:974]: This came from `SysBasic.syserr` and I think the intent was for it to be an opaque type.

**Comment by daviditen** [comment]
Thanks! I made the first two changes you suggested and replied to the third.

**Review by dlongnecke-cray — APPROVED** [review]

**Pull request merged**